src/Transports/MassTransit.EventHubIntegration/EventHubIntegration/Checkpoints/PartitionOffset.cs

```diff
@@ -0,0 +1,34 @@
+namespace MassTransit.EventHubIntegration.Checkpoints
+{
+    using Azure.Messaging.EventHubs.Processor;
+
+
+    // ReSharper disable NotAccessedField.Local
+    public readonly struct PartitionOffset
+    {
+        readonly string _eventHubName;
+        readonly string _partitionId;
+        readonly long _offset;
+
+        public PartitionOffset(ProcessEventArgs args)
+            : this(args.Partition.EventHubName, args.Partition.PartitionId, args.Data.Offset)
+        {
+        }
+
+        PartitionOffset(string eventHubName, string partitionId, long offset)
+        {
+            _eventHubName = eventHubName;
+            _partitionId = partitionId;
+            _offset = offset;
+        }
+    }
+
+
+    static class ProcessEventArgsExtensions
```

Comment:
If you're going this far, just create operator implicit and get rid of the duplicate constructor as well. Then you can just use assignments.